src/org/whstsa/library/api/impl/library/Library.java

```diff
@@ -130,7 +136,10 @@ public UUID getID() {
     }
 
     @Override
-    public ICheckout reserveBook(IMember member, IBook book) throws BookNotRegisteredException {
+    public ICheckout reserveBook(IMember member, IBook book) throws BookNotRegisteredException, OutOfStockException {
+        if (this.bookQuantity.get(book) == 0 || (this.getCheckouts().get(book) == null && this.getCheckouts().get(book).size() == this.bookQuantity.get(book))) {
```

Comment:
`this.getCheckouts().get(book) == null` should be `this.getCheckouts().get(book) != null` otherwise there will be a `NullPointerException` as the if statement will call methods on a null reference